refactor: 'non-primitive cast: `i32` as `()`' in `005_sphincs_PQCgenKAT_sign_blake_128f_simple`

To reproduce:

```sh
git clone https://github.com/immunant/c2rust.git
git clone https://github.com/DARPA-TRACTOR-Program/Test-Corpus.git
C2RUST_DIR="$PWD/c2rust"
(cd "$C2RUST_DIR" && cargo build --release)
export PATH="$C2RUST_DIR/target/release:$PATH"
cd Test-Corpus/Public-Tests/P01_sphincs_plus/005_sphincs_PQCgenKAT_sign_blake_128f_simple/test_case
cmake -S . -B build -G Ninja
bear -- cmake --build build/ -- driver
c2rust transpile compile_commands.json --binary PQCgenKAT_sign --emit-build-files --c2rust-dir "$C2RUST_DIR" --reorganize-definitions --disable-refactoring
c2rust refactor --cargo --rewrite-mode inplace rename_unnamed
c2rust refactor --cargo --rewrite-mode inplace reorganize_definitions
```

Error:

```shell
error[E0605]: non-primitive cast: `i32` as `()`
   --> app/src/sign.rs:105:5
    |
105 | /     crate::app::src::rng::randombytes(
106 | |         &raw mut seed as *mut ::core::ffi::c_uchar,
107 | |         (3 as ::core::ffi::c_int * crate::params_sphincs_haraka_128s_h::SPX_N) as ::core::ffi::c_ulonglong,
108 | |     ) as ();
    | |___________^ an `as` expression can only be used to convert between primitive types or to coerce to a specific trait object

error[E0605]: non-primitive cast: `i32` as `()`
   --> app/src/sign.rs:157:5
    |
157 | /     crate::app::src::rng::randombytes(
158 | |         &raw mut optrand as *mut ::core::ffi::c_uchar,
159 | |         crate::params_sphincs_haraka_128s_h::SPX_N as ::core::ffi::c_ulonglong,
160 | |     ) as ();
    | |___________^ an `as` expression can only be used to convert between primitive types or to coerce to a specific trait object

For more information about this error, try `rustc --explain E0605`.
error: could not compile `c2rust_out` due to 2 previous errors
thread 'main' panicked at 'Could not rebuild crate: 1 job failed', c2rust-refactor/src/lib.rs:379:38
stack backtrace:
   0:     0x767d1416cc60 - std::backtrace_rs::backtrace::libunwind::trace::h60bf900414c5dcf2
                               at /rustc/d394408fb38c4de61f765a3ed5189d2731a1da91/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   1:     0x767d1416cc60 - std::backtrace_rs::backtrace::trace_unsynchronized::habfc3ef92583c158
                               at /rustc/d394408fb38c4de61f765a3ed5189d2731a1da91/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x767d1416cc60 - std::sys_common::backtrace::_print_fmt::hae196fe4153d0b63
                               at /rustc/d394408fb38c4de61f765a3ed5189d2731a1da91/library/std/src/sys_common/backtrace.rs:66:5
   3:     0x767d1416cc60 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h08f9c50f850514f5
                               at /rustc/d394408fb38c4de61f765a3ed5189d2731a1da91/library/std/src/sys_common/backtrace.rs:45:22
   4:     0x767d141c862c - core::fmt::write::hcd15d2c673b5a9c1
                               at /rustc/d394408fb38c4de61f765a3ed5189d2731a1da91/library/core/src/fmt/mod.rs:1198:17
   5:     0x767d1415dca5 - std::io::Write::write_fmt::h417be4d25b915a22
                               at /rustc/d394408fb38c4de61f765a3ed5189d2731a1da91/library/std/src/io/mod.rs:1672:15
   6:     0x767d1416f941 - std::sys_common::backtrace::_print::h95ab226f37d571aa
                               at /rustc/d394408fb38c4de61f765a3ed5189d2731a1da91/library/std/src/sys_common/backtrace.rs:48:5
   7:     0x767d1416f941 - std::sys_common::backtrace::print::hc77fb9ed6b2eeb3d
                               at /rustc/d394408fb38c4de61f765a3ed5189d2731a1da91/library/std/src/sys_common/backtrace.rs:35:9
   8:     0x767d1416f941 - std::panicking::default_hook::{{closure}}::hca418bd67097eb22
                               at /rustc/d394408fb38c4de61f765a3ed5189d2731a1da91/library/std/src/panicking.rs:295:22
   9:     0x767d1416f60e - std::panicking::default_hook::h1df4f10c6ed28c53
                               at /rustc/d394408fb38c4de61f765a3ed5189d2731a1da91/library/std/src/panicking.rs:314:9
  10:     0x767d169f07a4 - rustc_driver[6a784e82872844b3]::DEFAULT_HOOK::{closure#0}::{closure#0}
  11:     0x767d14170176 - std::panicking::rust_panic_with_hook::hf1e7a1cb721a9823
                               at /rustc/d394408fb38c4de61f765a3ed5189d2731a1da91/library/std/src/panicking.rs:702:17
  12:     0x767d1416ffc7 - std::panicking::begin_panic_handler::{{closure}}::h573a00e95ce9340d
                               at /rustc/d394408fb38c4de61f765a3ed5189d2731a1da91/library/std/src/panicking.rs:588:13
  13:     0x767d1416d184 - std::sys_common::backtrace::__rust_end_short_backtrace::h597b7803117e52fd
                               at /rustc/d394408fb38c4de61f765a3ed5189d2731a1da91/library/std/src/sys_common/backtrace.rs:138:18
  14:     0x767d1416fcf2 - rust_begin_unwind
                               at /rustc/d394408fb38c4de61f765a3ed5189d2731a1da91/library/std/src/panicking.rs:584:5
  15:     0x767d141c5103 - core::panicking::panic_fmt::h62ccf03c8a8a51b5
                               at /rustc/d394408fb38c4de61f765a3ed5189d2731a1da91/library/core/src/panicking.rs:142:14
  16:     0x767d141c5463 - core::result::unwrap_failed::hff48f82f03d418ae
                               at /rustc/d394408fb38c4de61f765a3ed5189d2731a1da91/library/core/src/result.rs:1814:5
  17:     0x6223c3744084 - c2rust_refactor::main_impl::h2abbaede1af1b202
  18:     0x6223c3b30542 - rustc_driver::catch_fatal_errors::h650f4a5d11a80904
  19:     0x6223c3742c67 - c2rust_refactor::lib_main::h83b71f278f1eb1fd
  20:     0x6223c32c448d - c2rust_refactor::main::h9bcfd9f425787ff8
  21:     0x6223c32c4b33 - std::sys_common::backtrace::__rust_begin_short_backtrace::h00866a21418e4c51
  22:     0x6223c32c5039 - std::rt::lang_start::{{closure}}::hf976110e62d0b5c0
  23:     0x767d1414e15a - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h02e5de65baee3e0a
                               at /rustc/d394408fb38c4de61f765a3ed5189d2731a1da91/library/core/src/ops/function.rs:280:13
  24:     0x767d1414e15a - std::panicking::try::do_call::h928f62916d7b4bf7
                               at /rustc/d394408fb38c4de61f765a3ed5189d2731a1da91/library/std/src/panicking.rs:492:40
  25:     0x767d1414e15a - std::panicking::try::he5e8d9bda1fee0d9
                               at /rustc/d394408fb38c4de61f765a3ed5189d2731a1da91/library/std/src/panicking.rs:456:19
  26:     0x767d1414e15a - std::panic::catch_unwind::hbdacb0480f3c3a44
                               at /rustc/d394408fb38c4de61f765a3ed5189d2731a1da91/library/std/src/panic.rs:137:14
  27:     0x767d1414e15a - std::rt::lang_start_internal::{{closure}}::h9307a76627b54f4f
                               at /rustc/d394408fb38c4de61f765a3ed5189d2731a1da91/library/std/src/rt.rs:128:48
  28:     0x767d1414e15a - std::panicking::try::do_call::h0baecbe86a821dd2
                               at /rustc/d394408fb38c4de61f765a3ed5189d2731a1da91/library/std/src/panicking.rs:492:40
  29:     0x767d1414e15a - std::panicking::try::h6c0869befc863f62
                               at /rustc/d394408fb38c4de61f765a3ed5189d2731a1da91/library/std/src/panicking.rs:456:19
  30:     0x767d1414e15a - std::panic::catch_unwind::h135074234245f66c
                               at /rustc/d394408fb38c4de61f765a3ed5189d2731a1da91/library/std/src/panic.rs:137:14
  31:     0x767d1414e15a - std::rt::lang_start_internal::h498f9556b87c8e5f
                               at /rustc/d394408fb38c4de61f765a3ed5189d2731a1da91/library/std/src/rt.rs:128:20
  32:     0x6223c32c4b22 - main
  33:     0x767d1342a1ca - __libc_start_call_main
                               at ./csu/../sysdeps/nptl/libc_start_call_main.h:58:16
  34:     0x767d1342a28b - __libc_start_main_impl
                               at ./csu/../csu/libc-start.c:360:3
  35:     0x6223c32c0a75 - _start
  36:                0x0 - <unknown>

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.65.0-nightly (d394408fb 2022-08-07) running on x86_64-unknown-linux-gnu

query stack during panic:
end of query stack
```

--- a/c2rust-refactor/src/transform/externs.rs
+++ b/c2rust-refactor/src/transform/externs.rs
@@ -187,6 +187,13 @@ fn make_cast<'a, 'tcx>(
 
     if needs_transmute {
         mk().call_expr(mk().path_expr(vec!["", "core", "mem", "transmute"]), vec![expr])
+    } else if ty.is_unit() {
+        // `{ let _ = expr; }` since the context expects a unit tuple.
+        // `let` is technically a statement but its value is `()` which
+        // is exactly what we need.
+        mk().block_expr(mk().block(vec![
+            mk().local_stmt(P(mk().local(mk().wild_pat(), None::<P<ast::Ty>>, Some(expr)))),
+        ]))
     } else {
         let expr = if let ExprKind::AddrOf(_, mutability, _) = expr.kind {
             // We have to cast to *T where &T is the type of expr before casting

